lib/generateAndStoreSuggestions.ts

```diff
@@ -99,39 +103,71 @@ export async function generateAndStoreSuggestions(
       }
     }
 
-    const parsedResponse = JSON.parse(jsonContent);
-
-    // Process each suggestion with Amazon data
-    for (const suggestion of parsedResponse) {
-      const amazonData = await getAmazonImage(suggestion.title);
-
-      const cleanSuggestion = {
-        title: String(suggestion.title),
-        price: String(suggestion.price),
-        description: String(suggestion.description),
-        matchReasons: Array.isArray(suggestion.matchReasons)
-          ? suggestion.matchReasons.map(String)
-          : [],
-        matchScore: Number(suggestion.matchScore),
-        imageUrl: amazonData.imageUrl || null,
+    const parsedResponse: IGeneratedSuggestionRaw[] = ((): IGeneratedSuggestionRaw[] => {
+      const raw = JSON.parse(jsonContent) as unknown[];
+      return raw.map((item) => {
+        const obj = item as Record<string, unknown>;
```

Comment:
refactor (nit): might as well just cast once. Runtime validation might make sense too. We don't have Zod, but you could use type guards e.g. 

```
function isSuggestionLike(item: unknown): item is Record<string, unknown> {
  return typeof item === 'object' && item !== null;
}
```


```suggestion
      const raw = JSON.parse(jsonContent) as Record<string, unknown>[];
      return raw.map((item) => {
        const obj = item; // probably don't need obj intermediary var since it was for casting
```